flink-yarn/src/main/java/org/apache/flink/yarn/YarnResourceManager.java

```diff
@@ -527,10 +524,9 @@ private FinalApplicationStatus getYarnStatus(ApplicationStatus status) {
 	 * Request new container if pending containers cannot satisfy pending slot requests.
 	 */
 	private void requestYarnContainerIfRequired() {
-		int requiredTaskManagerSlots = getNumberRequiredTaskManagerSlots();
-		int pendingTaskManagerSlots = numPendingContainerRequests * numSlotsPerTaskManager;
+		int requiredTaskManagers = getNumberRequiredTaskManagers();
 
-		if (requiredTaskManagerSlots > pendingTaskManagerSlots) {
+		if (requiredTaskManagers > numPendingContainerRequests) {
```

Comment:
ditto here as in `KubernetesResourceManager `

Reply:
Same here.